### ISSUE TITLE
Source code for jpsxdec-stupid-edition.jar

### ISSUE THREAD

**Issue by m35** [ArcaneAria/Translation_Scripts#1]
Could you share the Java code that was changed in the jpsxdec-stupid-edition.jar ?
I'm curious about it, but also because [the license](https://github.com/m35/jpsxdec/blob/readme/.github/LICENSE.md) requires it.
Thank you

**Comment by SnowyAria** [comment]
I'm sorry about that, I didn't think about the ramifications of including the jar files for convenience. As that test jar didn't do anything useful, I went ahead and removed it and the other older jars, instead updating the README to say to grab the latest from the official repo.

Now, I might be a bit hazy on the details since this was two year ago, but I found the code changes and my notes. As this SRT tool is for translation, I wanted to try reducing the patch sizes by changing as little of the STR file as possible. The changes to the jar were motivated by the partial frame replace feature not behaving correctly at the time with a region and transparency. Using python, I tried making a bounding box that fit snuggly around the subtitles. The goal was to change as little of the frame as possible.

![image](https://github.com/user-attachments/assets/82e219b8-3f59-492c-af1b-47b6f06e26ec)

My notes at the time: "So from what I can tell, it gets the changed blocks to insert, but it tries to force the changed section to fit in the existing byte space. If the inserted bytes are too large (almost always are), it loops and compresses them until it's small enough to fit. Then that results in pixel square nonsense
I'm not really sure if it has to obey that size or if it just is for some reason, so that's something to investigate. Otherwise reducing the quality of the input images might help..." I'm assuming the code changes I made tried to lazily handle this, but I don't fully remember honestly.

I was able to make it work in the end with these files changed from the jar - [jPSXdec-partial-replace.zip](https://github.com/user-attachments/files/17626050/jPSXdec-partial-replace.zip) . Overall the changes were pretty hacky and almost certainly error-prone corner cutting (hence stupid-edition.jar), so I honestly don't know if these even have any value being looked at, but I'll include them just in case.

In the end, I was able to reduce the resulting xdelta of just the test video from 1.87mb to 1.56mb, but with a little more artifacting around the region.

Full replace:
![Ancient_Roman_video_full_replace](https://github.com/user-attachments/assets/ac73b4c0-82ce-4652-97cf-1704958082f4)

Partial replace:
![Ancient_Roman_video_partial_replace](https://github.com/user-attachments/assets/f58cdf63-4016-42f9-8828-7ca983bdac90)

It's subtle and hard to see from these screenshots, but there's extra shimmer in the bounding box of the subtitles.

Overall, I abandoned the effort after the small reduction in file size versus making things look worse. If I spent a bit more time learning how your code worked and doing it right, maybe I could have made it look better, but we shifted over from hard subs to soft subs before long for most of our projects, so this SRT tool got overtaken by events, really.

Anyway, wanted to give you a full explanation for the situation. Sorry again about the license violation, I didn't think that through. Thank you for all the work you do on jPSXdec!
